### ISSUE TITLE
Suspense fallback is triggered when no request is made

### ISSUE THREAD

**Issue by drasco** [coinbase/rest-hooks#1438]
**React version** 17.0.2

**Concurrent mode** yes/no

**Rest Hooks version** 6.1.6

**Describe the bug**
A map generated card list flickers (because the fallback component is rendered), when a sever side search filter query param is applied, even if the data is in cache

**To Reproduce**
This code, using suspense, causes flicker, especially if the browsing session is over a remote terminal server
i.e "Loading" gets triggered even though the cache is "fresh" and no network request is made
```
export default function  MachineList({sortBy}: Props){    
    const critArray: string[] = [];
    const {filter,actions} = useFilterStore();
    for (var key in filter.criteria) critArray.push( key);
    const machines = useResource(MachineResource.list(), {sortBy, criteria: critArray.join(',')});
    return (
        <Card.Group   >            
            {machines.map(machine => <MachineCard key={machine.pk()} machine={machine} /> )}
        </Card.Group>
    )
}
```

**Expected behavior**
Using rest-hooks/legacy useStatefulResource, this does not occur
```export default function  MachineList({sortBy}: Props){
    const critArray: string[] = [];
    const {filter,actions} = useFilterStore();
    for (var key in filter.criteria) critArray.push( key);
    const { data, loading, error } = useStatefulResource( MachineResource.list(), {sortBy, criteria: critArray.join(',')} );
    if (error) return <div>Error {error.status}</div>
    if (loading) {
        console.log("loading");
        return <div>Loading</div>;
    }
    else return (
        <Card.Group>            
            {data?.map(machine => <MachineCard key={machine.pk()} machine={machine} /> ) }
        </Card.Group>
    )
}
```

**Additional context**
Ideally it would be nice to have an option where suspense can be disabled even if a network request is made. Perhaps show the Loading only after the request has taken 200ms+

**Comment by drasco** [comment]
Side note, is it possible to have more examples on the topic of server side filtering?

When combined with the cache, it really is a great feature - it makes client side filtering quite redundant.

**Comment by ntucker** [comment]
If you're down to do a screen share pair - join our discord https://discord.gg/35nb8Mz and I'm happy to take a look tomorrow.

**Comment by ntucker** [comment]
@drasco Still down to pair if you are. :)

**Comment by drasco** [comment]
Hey Nathan, terribly sorry - i'm back on the main .net app for a while now before the web page gets any attention. I can barely even remember how react works now!

**Comment by ntucker** [comment]
Closing this for now, but feel free to reach out on discord if you see the problem again. Also React 18 final is out now :)
